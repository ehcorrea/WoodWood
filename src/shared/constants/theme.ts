import { rfvalue, rhvalue, rwvalue } from '../utils/responsive/responsive';

const COLORS_SCHEMES = {
  light: {
    primary: {
      main: '#EA5B0C',
      80: '#EF7A39',
      50: '#F4A273',
      20: '#F9D2B6',
    },
    secondary: {
      main: '#004ABE',
      80: '#336ECC',
      50: '#7FA5E0',
      20: '#CCD9F5',
    },
  },
};

export const THEME = {
  colors: {
    black: {
      main: '#1A1D26',
      80: '#2A2F3D',
      50: '#4D5364',
      20: '#6E7489',
    },
    white: {
      main: '#FFFFFF',
      80: '#FFFFFF',
      50: '#FFFFFF',
      20: '#FFFFFF',
    },
    gray: {
      main: '#9A9FAE',
      80: '#A8ACB9',
      50: '#C4C7D0',
      20: '#EBEBEB',
    },
    error: {
      main: '#F30000',
      80: '#F34C4C',
      50: '#F38585',
      20: '#F3BEBE',
    },
    transparent: {
      main: 'transparent',
      80: 'transparent',
      50: 'transparent',
      20: 'transparent',
    },
    ...COLORS_SCHEMES.light,
  },
  fonts: {
    weight: {
      light: 'SofiaPro-Light',
      regular: 'SofiaPro-Regular',
      medium: 'SofiaPro-Medium',
      semibold: 'SofiaPro-SemiBold',
      bold: 'SofiaPro-Bold',
    },
    size: {
      small: rfvalue(12).toString(),
      medium: rfvalue(15).toString(),
      large: rfvalue(18).toString(),
      xlarge: rfvalue(24).toString(),
      huge: rfvalue(30).toString(),
      xhuge: rfvalue(36).toString(),
    },
    height: {
      small: rfvalue(12).toString(),
      medium: rfvalue(18).toString(),
      large: rfvalue(20).toString(),
      xlarge: rfvalue(30).toString(),
      huge: rfvalue(50).toString(),
      xhuge: rfvalue(55).toString(),
    },
  },
  rfvalue,
  rwvalue,
  rhvalue,
} as const;

export type ThemeType = typeof THEME;
