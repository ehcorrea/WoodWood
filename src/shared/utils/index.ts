export * from './carrinho/carrinho';
export * from './responsive/responsive';
