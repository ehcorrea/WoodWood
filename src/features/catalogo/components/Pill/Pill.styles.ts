import styled, { css } from '@emotion/native';
import { Theme } from '@emotion/react';

import { Text as TextComp } from '@/shared/components';

const wraperModifiers = {
  selecionado: (theme: Theme) => css`
    background-color: ${theme.colors.primary.main};
  `,
};

const textModifiers = {
  selecionado: (theme: Theme) => css`
    color: ${theme.colors.white.main};
  `,
};

type PillProps = {
  selecionado: boolean;
};

export const Wraper = styled.View<PillProps>`
  ${({ theme, selecionado }) => css`
    align-items: center;
    background-color: ${theme.colors.primary[50]};
    border-radius: 20px;
    height: 30px;
    justify-content: center;
    padding-horizontal: 10px;

    ${selecionado && wraperModifiers.selecionado(theme)}
  `}
`;

export const Text = styled(TextComp)<PillProps>`
  ${({ theme, selecionado }) => css`
    color: ${theme.colors.black.main};

    ${selecionado && textModifiers.selecionado(theme)}
  `}
`;
