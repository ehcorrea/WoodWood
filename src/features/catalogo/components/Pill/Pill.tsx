import * as S from './Pill.styles';

type PillProps = {
  label: string;
  categoriaSelecionada: string | undefined;
};

export function Pill({ label, categoriaSelecionada }: PillProps) {
  const selecionado = label === categoriaSelecionada;

  return (
    <S.Wraper selecionado={selecionado} testID="wraper-pill">
      <S.Text selecionado={selecionado} weight="semibold">
        {label}
      </S.Text>
    </S.Wraper>
  );
}
