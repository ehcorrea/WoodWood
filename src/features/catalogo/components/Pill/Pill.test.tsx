import { render } from '@/test/utils';
import { Pill } from './Pill';
import { THEME } from '@/shared/constants/theme';

const setup = (categoriaSelecionada = 'category') => {
  const container = render(
    <Pill categoriaSelecionada={categoriaSelecionada} label="category" />
  );
  const wrapper = container.getByTestId('wraper-pill');
  const text = container.getByText('category');

  return { container, wrapper, text };
};

describe('<Pill/>', () => {
  describe('quando renderizado', () => {
    test('e props categoriaSelecionada igual a label', () => {
      const { wrapper, text, container } = setup();
      expect(wrapper).toHaveStyle({
        backgroundColor: THEME.colors.primary.main,
      });
      expect(text).toHaveStyle({
        color: THEME.colors.white.main,
      });
      expect(container).toMatchSnapshot();
    });
    test('e props categoriaSelecionada diferente da label', () => {
      const { wrapper, text, container } = setup('category-2');
      expect(wrapper).toHaveStyle({
        backgroundColor: THEME.colors.primary[50],
      });
      expect(text).toHaveStyle({
        color: THEME.colors.black.main,
      });
      expect(container).toMatchSnapshot();
    });
  });
});
