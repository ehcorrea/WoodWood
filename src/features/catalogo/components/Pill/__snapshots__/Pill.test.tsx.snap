// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pill/> quando renderizado e props categoriaSelecionada diferente da label 1`] = `
<View
  selecionado={false}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#F4A273",
        "borderBottomLeftRadius": 20,
        "borderBottomRightRadius": 20,
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 30,
        "justifyContent": "center",
        "paddingHorizontal": 10,
      },
      undefined,
    ]
  }
  testID="wraper-pill"
>
  <Text
    selecionado={false}
    style={
      [
        {
          "color": "#1A1D26",
          "fontFamily": "SofiaPro-SemiBold",
          "fontSize": 21.5,
          "lineHeight": 25.5,
        },
        [
          {
            "color": "#1A1D26",
          },
          undefined,
        ],
      ]
    }
    weight="semibold"
  >
    category
  </Text>
</View>
`;

exports[`<Pill/> quando renderizado e props categoriaSelecionada igual a label 1`] = `
<View
  selecionado={true}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#EA5B0C",
        "borderBottomLeftRadius": 20,
        "borderBottomRightRadius": 20,
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 30,
        "justifyContent": "center",
        "paddingHorizontal": 10,
      },
      undefined,
    ]
  }
  testID="wraper-pill"
>
  <Text
    selecionado={true}
    style={
      [
        {
          "color": "#1A1D26",
          "fontFamily": "SofiaPro-SemiBold",
          "fontSize": 21.5,
          "lineHeight": 25.5,
        },
        [
          {
            "color": "#FFFFFF",
          },
          undefined,
        ],
      ]
    }
    weight="semibold"
  >
    category
  </Text>
</View>
`;
